ravedude/src/avrdude/mod.rs

```diff
@@ -17,6 +17,50 @@ pub struct Avrdude {
 }
 
 impl Avrdude {
+    pub fn require_min_ver((req_major, req_minor): (u8, u8)) -> anyhow::Result<()> {
+        let output = process::Command::new("avrdude")
+            .arg("-?")
+            .output()
+            .context("Failed to start and gather avrdude output.")?;
+        let stderr: &str =
+            std::str::from_utf8(&output.stderr).context("Avrdude's stderr contains non-utf8.")?;
+        let version: &str = stderr
+            .split("version")
+            .last()
+            .context("Unable to obtain avrdude's version string")?
+            .trim();
+        let (version, _) = version
+            .split_once('-')
+            .context("Unable to obtain avrdude's version string")?
+            .to_owned();
+        let (major, minor) = version.split_once('.').unwrap();
+        let major = major
+            .parse::<u8>()
+            .context("Unable to parse major version number.")?;
+        let minor = minor
+            .parse::<u8>()
+            .context("Unable to parse minor version number.")?;
+        if major < req_major {
+            bail!(
+                "Avrdude does not meet minimum version requirements. v{}.{} was found while v{}.{} or greater is required.",
```

Comment:
Just as an idea: Can we point to the official download page for avrdude in this error message?  Just so people are immediately guided in the right direction?  Especially for the Windows users, not having them download avrdude from the first link off Google might be good.

Reply:
Sure, no problem!

Reply:
Thank you. I have found your suggestions and guidance very helpful and appreciate you took the time.
Just pushed an new rendition of the PR hopefully addressing most concerns. 
I am not sure when to resolve a conversation. Intuitively I'd think 'when the outcome is satisfactory'?

Reply:
Sorry, it seems I botched. I tried to revert 'digital' spelling fix, but it seems I changed two instances. I cannot find the simple way of reverting a single hunk in a patch, so I did it manually. Sigh.

Reply:
I did and I fixed it.